### ISSUE TITLE
HTMLElement life-cycle callbacks

### ISSUE THREAD

**Issue by adaliszk** [Draccoz/twc#121]
When I try to use the HTMLElement life-cycle callbacks then I got errors by TypeScript:
```
TS2339: Property 'connectedCallback' does not exist on type 'PolymerElement'.
TS2339: Property 'disconnectedCallback' does not exist on type 'PolymerElement'.
```

Polymer repository has a typescript definition, which might help to implement or ask to release that as a separate: https://github.com/Polymer/polymer/blob/typescript/polymer.d.ts

**Comment by Draccoz** [comment]
This is not as simple as one would think. The types you linked to do not work out of the box, and last time I talked with the team about it, they are not (at least for now) going to do anything about those types, as they concentrate on making a generator to automatically generate types. Typings under `polymer.d.ts` is a copy of previous types on Polymer repo, tweaked a bit by me so they work nicely. I have also tried to fix those types and publish on `@types`, but they refused to merge my types as I do not own Polymer itself... Ended up with types on [this](https://github.com/Draccoz/polymer2-types) repo (published on npm [here](https://www.npmjs.com/package/polymer2-types)). After that I had 3 months of no time for twc, and now forgot to mention that, but will update twc and how to use those types soon (after I am done with making mixins work).
